services/meta/data.go

```diff
@@ -371,6 +372,9 @@ func (data *Data) CreateShardGroup(database, policy string, timestamp time.Time)
 	sgi.ID = data.MaxShardGroupID
 	sgi.StartTime = timestamp.Truncate(rpi.ShardGroupDuration).UTC()
 	sgi.EndTime = sgi.StartTime.Add(rpi.ShardGroupDuration).UTC()
+	if sgi.EndTime.After(models.MaxNanoTime) {
+		sgi.EndTime = models.MaxNanoTime
```

Comment:
In here also need to do `sgi.StartTime = sgi.EndTime.Add(-models.MaxNanoTime.UnixNano())` or similar.